flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FWC.js

```diff
@@ -420,4 +460,136 @@ class A32NX_FWC {
+            if (!this.warningPriority) {
+                SimVar.SetSimVarValue("L:Generic_Master_Warning_Active", "Bool", false);
+            }
+        } else {
+            if (this.APDisconnectedBySidestick && !this.AutopilotWarningCanceled) {
+                // AUTOPILOT DISCONNECTED BY SIDESTICK
+                SimVar.SetSimVarValue("L:A32NX_AUTOPILOT_DISCONNECT_BY_SIDESTICK", "Bool", true);
+                SimVar.SetSimVarValue("L:Generic_Master_Warning_Active", "Bool", true);
+                this.apdeltaTime += _deltaTime;
+
+                // Stop warning when press Master Warning or pass 3 sec
+                if (this.warningPressed || (this.apdeltaTime / 1000) >= 3) {
+                    this.AutopilotWarningCanceled = true;
+                    SimVar.SetSimVarValue("L:A32NX_AUTOPILOT_DISCONNECT_BY_SIDESTICK", "Bool", false);
+
+                    // If there is a Prioirty Warning, DO NOT STOP WAWRNING
+                    if (!this.warningPriority) {
+                        SimVar.SetSimVarValue("L:Generic_Master_Warning_Active", "Bool", false);
+                    }
+                    this.apdeltaTime = 0;
+                }
+            } else if (!this.APDisconnectedBySidestick && !this.AutopilotWarningCanceled) {
+                // AUTOPILOT DISCONNECTED BY FCU
+                SimVar.SetSimVarValue("L:A32NX_AUTOPILOT_DISCONNECT_BY_FCU", "Bool", true);
+                SimVar.SetSimVarValue("L:Generic_Master_Warning_Active", "Bool", true);
+
+                // Stop warning when press Master Warning
+                if (this.warningPressed) {
+                    this.AutopilotWarningCanceled = true;
+
+                    // If there is a Prioirty Warning, DO NOT STOP WAWRNING
+                    if (!this.warningPriority) {
+                        SimVar.SetSimVarValue("L:Generic_Master_Warning_Active", "Bool", false);
+                    }
+                }
+            }
+        }
+    }
+
+    _checkLandingGear(_deltaTime) {
+        const radioHeight = SimVar.GetSimVarValue("RADIO HEIGHT", "Feet");
+        const eng1N1 = SimVar.GetSimVarValue("ENG N1 RPM:1", "Percent");
+        const eng2N1 = SimVar.GetSimVarValue("ENG N1 RPM:2", "Percent");
+        const isLandingGearLockedDown = SimVar.GetSimVarValue("GEAR POSITION:0", "Enum") > 0.9;
+        const isTogaFlexMct1 = SimVar.GetSimVarValue("GENERAL ENG THROTTLE MANAGED MODE:1", "number") > 4;
+        const isTogaFlexMct2 = SimVar.GetSimVarValue("GENERAL ENG THROTTLE MANAGED MODE:2", "number") > 4;
+        const flapPosition = SimVar.GetSimVarValue("FLAPS HANDLE INDEX", "Number");
+        const isEngine1Shutdown = (SimVar.GetSimVarValue("TURB ENG N1:1", "Percent") < 15 || SimVar.GetSimVarValue("FUELSYSTEM VALVE SWITCH:1", "Bool") == 0) && !Simplane.getIsGrounded();
+        const isEngine2Shutdown = (SimVar.GetSimVarValue("TURB ENG N1:2", "Percent") < 15 || SimVar.GetSimVarValue("FUELSYSTEM VALVE SWITCH:2", "Bool") == 0) && !Simplane.getIsGrounded();
```

Comment:
**issue**: SimVar reads are expensive. Avoid unnecessary simvar reads, as most of the time not all variables will be necessary to determine the result.
**suggestion**: `return` after each condition. Don't use `else if` but `if` statements, reading the necessary variables just before the `if` statement.